### PR TITLE
[core] Optimize tags system table when specifying tag_name

### PR DESCRIPTION
**Pull request by herefree**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

**Review by JingsongLi — APPROVED** [review]
+1

File: paimon-core/src/main/java/org/apache/paimon/table/system/TagsTable.java [hunk 1]
```diff
@@ -202,10 +217,23 @@ public RecordReader<InternalRow> createReader(Split split) {
...
+
             Map<String, Tag> nameToSnapshot = new LinkedHashMap<>();
-            for (Pair<Tag, String> tag : tags) {
-                nameToSnapshot.put(tag.getValue(), tag.getKey());
+
+            if (predicate != null
+                    && predicate.function() instanceof Equal
+                    && predicate.literals().get(0) instanceof BinaryString) {
+                String equalValue = predicate.literals().get(0).toString();
+                Tag tag = tagManager.tag(equalValue);
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/table/system/TagsTable.java:229]: Use `tagExists` first.

File: paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java [hunk 2]
```diff
@@ -382,4 +383,13 @@ private int findIndex(Snapshot taggedSnapshot, List<Snapshot> taggedSnapshots) {
                         "Didn't find tag with snapshot id '%s'.This is unexpected.",
                         taggedSnapshot.id()));
     }
+
+    /** Read tag for tagName. */
+    public Tag tag(String tagName) {
+        try {
+            return Tag.safelyFromPath(fileIO, tagPath(tagName));
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/utils/TagManager.java:390]: use Tag.fromPath

**Comment by herefree** [comment]
@JingsongLi thanks your comment,I have finished the modification.

**Comment by herefree** [comment]
image
flink UTCase passed in my workflow.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**